tools/cli/commands/release.js

```diff
@@ -260,19 +260,20 @@ export async function getReleaseVersion( argv ) {
 		return `${ stableVersion }-beta`;
 	}
 
-	// Append '-alpha' if necessary.
-	if ( argv.a || argv[ 'dev-version' ] ) {
-		// Jetpack uses additional versioning for dev/atomic in the form of x.y-a.z
-		if ( argv.project === 'plugins/jetpack' ) {
-			if ( alphaVersion ) {
-				alphaVersion = await getVersionBump( alphaVersion );
-				potentialVersion = `${ stableVersion }-${ alphaVersion }`;
+	// Handle alpha/dev-release version if necessary.
+	if ( argv.a || argv.devRelease ) {
+		// Check if dev-releases is specified in project's composer.json
+		const hasDevReleases = await readComposerJson( argv.project ).extra[ 'dev-releases' ];
```

Comment:
I was going to suggest using optional chaining to handle the case where `extra` isn't set in composer.json, like
```js
		const hasDevReleases = await readComposerJson( argv.project ).extra?.[ 'dev-releases' ];
```
except the `esm` package we're using breaks on that syntax and [isn't being updated](https://href.li/?https://github.com/standard-things/esm/issues/866). Sigh.

Reply:
Ah that woulda been slick.